ygong/mosaic/submit.py

```diff
@@ -204,29 +214,29 @@ def on_button_clicked():
             summary = _get_run_summary(run, mlflow_experiment_name)
             display(HTML(summary.to_html(escape=False)))
         button.on_click(on_button_clicked)
-    _display_run_summary(_get_run_summary(run, mlflow_experiment_name), button)
-    run = _wait_for_run_status(run, RunStatus.RUNNING)
-    _display_run_summary(_get_run_summary(run, None), button)
-
-    try_count = 0
-    while try_count < 10:
-        try_count += 1
-        time.sleep(20)
-        try:
-            run = get_run(run)
-            if run.status.is_terminal():
-                logger.debug(f"run {run_name} is in terminal state. Status {run.status}")
-                break
-            summary = _get_run_summary(run, mlflow_experiment_name)
-            _display_run_summary(summary, button)
-            break
-        except ValueError as e:
-             logger.debug(f"waiting for the MLFLow experiment run to be ready, run status {run.status}, error {e}")
-             pass
+    # _display_run_summary(_get_run_summary(run, mlflow_experiment_name), button)
```

Comment:
@ygong1 I commented these lines out other wise the display status won't get refreshed on time so it would be impossible to demonstrate preemption and system failure retry

Reply:
yeah, I'm refactoring the logic to watch on run status changes, will update soon

Reply:
I have refactored the displaying logic, it should be able display all status changes immediately now, please take another try, it should work now